ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -702,8 +702,11 @@ private static void setSwitchCurrentLimit(UcteLine ucteLine, Switch sw) {
         }
     }
 
-    private static UctePowerPlantType energySourceToUctePowerPlantType(EnergySource energySource) {
-        switch (energySource) {
+    private static UctePowerPlantType energySourceToUctePowerPlantType(Generator generator) {
+        if (generator.hasProperty(UcteImporter.UCTE_POWERPLANT_TYPE_KEY + generator.getId())) {
+            return UctePowerPlantType.valueOf(generator.getProperty(UcteImporter.UCTE_POWERPLANT_TYPE_KEY + generator.getId()));
+        }
```

Comment:
There is no good reason to make the property name dependent on the ID of the generator.